forms/gridfield/GridField.php

```diff
@@ -177,6 +174,11 @@ public function getConfig() {
 	public function setConfig(GridFieldConfig $config) {
 		$this->config = $config;
 
+		if (!$this->config->getComponentByType('GridState_Component')) {
+			$this->config->addComponent(new GridState_Component());
+			$this->state = new GridState($this);
```

Comment:
I think we need to keep it in the constructor and then ensure it's retained when a new config is assigned...